storage_controller/src/service.rs

```diff
@@ -5355,9 +5445,29 @@ impl Service {
             waiters = self
                 .await_waiters_remainder(waiters, SHORT_RECONCILE_TIMEOUT)
                 .await;
+
+            failpoint_support::sleep_millis_async!("sleepy-drain-loop");
         }
 
         while !waiters.is_empty() {
+            if cancel.is_cancelled() {
```

Comment:
This cancellation token we're checking: it looks like it's not a child of the service's cancellation token, so this will respect explicit cancellation but not service shutdown.  Let's fix that?

Reply:
Great point. Will fix.

Reply:
Done in [ff60248](https://github.com/neondatabase/neon/pull/8029/commits/ff602484856048d600133bcea23605b344600076)